src/library_exceptions.js

```diff
@@ -391,15 +392,61 @@ var LibraryExceptions = {
     {{{ makeThrow('ptr') }}}
   },
 
-#if !DISABLE_EXCEPTION_CATCHING
+#endif
+#if !DISABLE_EXCEPTION_CATCHING || EXCEPTION_HANDLING
+#if EXCEPTION_HANDLING
+  $getCppExceptionTag: function() {
+    return Module['asm']['__cpp_exception'];
+  },
+
+  $getCppExceptionThrownObject__deps: ['$getCppExceptionTag'],
+  $getCppExceptionThrownObject: function(ex) {
+    return ex.getArg(getCppExceptionTag(), 0);
+  },
+
+  $incrementExceptionRefcount__deps: ['__increment_wasm_exception_refcount', '$getCppExceptionThrownObject'],
+  $incrementExceptionRefcount: function(ptr) {
```

Comment:
The argument here is not a pointer but some kind of object right?

Reply:
Yeah right, this is confusing. I changed this to `obj`, and also changed function name `getCppExceptionThrownObject` to `getCppExceptionThrownValue` in the same file.